plugins/experimental/cookie_remap/cookiejar.cc

```diff
@@ -0,0 +1,273 @@
+   0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,  /* 90-9F */
+   0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,  /* A0-AF */
+   0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,  /* B0-BF */
+   0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,  /* C0-CF */
+   0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,  /* D0-DF */
+   0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,  /* E0-EF */
+   0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0   /* F0-FF */
+};
+
+bool
+CookieJar::create(const string &strCookie)
+{
+
+  if (strCookie.empty())
+  {
+    return false;
+  }
+
+  if (parse(strCookie, "; ", true, true) != 0)
+  {
+    return false;
+  }
+
+  return true;
+}
+
+int
+CookieJar::parse(const string &arg, const char *sepstr, bool val_check, bool mainElement)
+{
+  char *arg_copy;
+  char *cp;
+  char *key;
+
+  if ((arg_copy = strdup(arg.c_str())) == nullptr)
+    return -1;
+
+  cp = arg_copy;
+  for (key = strsep(&cp, sepstr); key != nullptr; key = strsep(&cp, sepstr))
+  {
+    int val_len;
+    char *val = strchr(key, '=');
+    char *addme = nullptr;
+
+    if (val)
+    {
+      /* split key and value */
+      *val++ = '\0';
+      val_len = strlen(val);
+      if (val_len > 0)
+      {
+	/* if we have DQUOTES around our value then drop them */
+
+	if (val_len > 1 && val[0] == '"' && val[val_len-1] == '"')
+	{
+	  val[val_len-1] = '\0';
+	  addme = val + 1;
+
+	  /* update the value length accordingly */
+
+	  val_len -= 2;
+	} 
+	else
+	{             
+	  addme = val;                 /* We have got a valid value eg: "YL=a" */
+	}
+
+	/* verify that the value is valid according to our configured
+	 * opton and possibly strip out invalid characters. */
+
+	if (val_check && verify_value(addme, val_len) != 0)
+	  continue;
+      }
+      else
+      { 
+	addme = "";                  /* Empty cookie case eg: "YL=" */
+      }
+    }
+    else /* If val is nullptr, then no need to add this kv pair to hashtable, skip this key */
+    {
+      continue;
+    }
+
+    /* we are going to verify the key name for our top level
+     * cookie names only so we'll use the val_check variables
+     * to know what we're processing */
+
+    if (val_check && verify_name(key) != 0)
+      continue;
+
+    if (mainElement)
+      addElement(key, addme);
+    else
+      addSubElement(key, addme);
+  }
+
+  free(arg_copy);
+  return 0;
+}
+
+void
+CookieJar::addElement(const char *key, const char *val)
+{
+  /* The insert method avoids duplicates */
+  CookieVal cval;
+  cval.m_val = val;
+  m_jar.insert(std::make_pair(key, cval));
+}
+
+void
+CookieJar::addSubElement(const char *key, const char *val)
+{
+  /* The insert method avoids duplicates */
+  m_currentVal->m_subelements.insert(std::make_pair(key, val));
+}
+
+int
+CookieJar::verify_value(char *val, int val_len)
+{
+  char *buf;
+  char *data_ptr = nullptr;
+  char data_buf[1024] = {0,};
+  int buf_len;
+  
+  if (val_len > (int) sizeof(data_buf) - 1)
```

Comment:
No C-style casts. Try `int(sizeof(data_buf))`.

Reply:
Better to do `static_cast<int>(sizeof(data_buf))`